### PR TITLE
docs: adr init wizard v2 and related improvements

### PR DESCRIPTION
**Pull request by aorumbayev**
Fixes (todo: include link to related minor enhancement spike)

### Init wizard v2
  - Outlines a new interactive flow for init wizard experience
### Command orchestration and `import contract` generator
  - Outlines new proposals around improving management on full-stack templates and generators
  - Outlines a new proposal to introduce more refined semantic around `project` command group that can aggregate static (and already project-centric) commands like `deploy`, `bootstrap` and allow defining any amount of extra user specific custom commands.

### CODE REVIEW

**Comment by aorumbayev** [comment]
@iskysun96 wasn't able to tag you directly by ccing here for visibility

**Review by robdmoore — APPROVED** [review]

File: docs/architecture-decisions/2024-01-23_init-wizard-v2.md [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+1. **Enhancements to the `Init` Wizard:**
+
+   - Improve user experience by making the wizard more intuitive and less reliant on Algorand-specific jargon.
+   - Streamline the `presets` concept to minimize user inputs and simplify the process.
+
+2. **Template Refinements:**
+
+   - Address potential complexities in maintaining the `fullstack` template with new `smart-contract` template combinations, including future `.NET` integration alongside `puya` compiler-compatible stacks.
+   - Implement `codespaces` configurations for simplified project setup in GitHub Codespaces.
+   - Consider unifying `add smart contract` generators in the `puya` and `beaker` templates into a single, generic generator managed by the CLI itself (suggestion added as part of revision 2 based on DevRel feedback).
```
Comment (neilcampbell) [inline docs/architecture-decisions/2024-01-23_init-wizard-v2.md:30]: Interested in understanding the problem this solves.

**Review by neilcampbell — APPROVED** [review]

File: docs/architecture-decisions/2024-01-23_init-wizard-v2.md [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+1. Break down the selected options into detailed work items.
+2. Implement the proposed changes in the AlgoKit CLI and templates.
+3. Update documentation and onboarding materials to reflect the new changes.
+4. Announce the updates to the community and DevRel, soliciting feedback for future enhancements.
+5. Tackle minor improvements that become feasible following the main proposal implementations.
+
+### Open Questions
+
+1. Should we implement a bidirectional query mechanism where selecting a smart contract prompts the option to add a frontend, and similarly, choosing a frontend first triggers a query about integrating a smart contract? This approach necessitates a minor modification in the wording of the initial query. In the current workflow, we don't prompt adding smart contracts part since this question is prominently positioned as the initial query in the wizard's sequence.
```
Comment (neilcampbell) [inline docs/architecture-decisions/2024-01-23_init-wizard-v2.md:210]: Also had this thought when reading the flowchart.

**Review by Loedn — APPROVED** [review]

File: docs/architecture-decisions/2024-01-23_init-wizard-v2.md (hunk 1 above)
Comment (aorumbayev) [reply docs/architecture-decisions/2024-01-23_init-wizard-v2.md]: @neilcampbell as discussed offline - suggestion certainly will improve aspects around simplifying template building experience and offloading some redundant logic from existing templates but will research a bit deeper and cover this as a follow up once all main pieces within the adr scope are implemented (cc @joe-p )

File: docs/architecture-decisions/2024-01-23_init-wizard-v2.md (hunk 2 above)
Comment (aorumbayev) [reply docs/architecture-decisions/2024-01-23_init-wizard-v2.md:210]: Its a simple change, let's see on initial feedback once this is a bit more polished with tests and docs and we can add bidirectional flow here as needed

**Pull request merged**